crates/core/src/model/scalar/mod.rs

```diff
@@ -93,6 +93,14 @@ impl<'s> ScalarCow<'s> {
         }
     }
 
+    /// Interpret as a string, if possible
+    pub fn to_string(self) -> Option<KStringCow<'s>> {
```

Comment:
This crate treats liquid as stringly-typed.  `ValueView::to_kstr` can be used get a string rendering of any value, borrowing if possible.  Would that work for your needs?

Reply:
> Would that work for your needs?

Almost, but no!

The `ValueView` impl for `ScalarCow` is as follows:

```rust
impl<'s> ValueView for ScalarCow<'s> {
  fn to_kstr(&self) -> KStringCow<'_> { /* etc. */ }
}
```

In other words, it's:

```rust
impl<'s> ValueView for ScalarCow<'s> {
  fn to_kstr<'a>(&'a self) -> KStringCow<'a> { /* etc. */ }
}
```

..but I need a `KStringCow<'s>`, not a `KStringCow<'a>`.

So, short of adding a lifetime parameter to `ValueView` itself (I tried it, tons of changes, didn't go well), we really do need a method on `ScalarCow` itself, this would also work:

```rust
    /// Interpret as a Cow str, borrowing if possible
    pub fn into_cow_str(self) -> Cow<'s, str> {
        match self {
            Self(ScalarCowEnum::Str(x)) => x.into_cow_str(),
            other => other.into_string().into_cow_str(),
        }
    }
```